firebase_admin/_auth_utils.py

```diff
@@ -323,3 +340,72 @@ def _build_error_message(code, exc_type, custom_message):
         exc_type and hasattr(exc_type, 'default_message')) else 'Error while calling Auth service'
     ext = ' {0}'.format(custom_message) if custom_message else ''
     return '{0} ({1}).{2}'.format(default_message, code, ext)
+
+
+def emulator_host():
+    return os.environ.get('FIREBASE_AUTH_EMULATOR_HOST')
+
+
+def is_emulator_enabled():
+    return bool(emulator_host())
+
+
+class AuthResourceUrlBuilder(object):
+    """Utility class to help building resource URLs."""
+
+    def __init__(self, project_id, version='v1'):
+        """The resource URL builder constructor.
+
+        Args:
+            project_id: The resource project ID.
+            version: The endpoint API version.
+        """
+        self.project_id = project_id
+        self.version = version
+        self._url_format = (
+            FIREBASE_AUTH_EMULATOR_BASE_URL_FORMAT.format(host=emulator_host())
+            if is_emulator_enabled() else FIREBASE_AUTH_BASE_URL_FORMAT)
+
+    def get_url(self, api='', **params):
+        """Returns the resource URL corresponding to the provided parameters.
+
+        Args:
+            api: The backend API name (optional).
+            **params: The optional additional parameters to substitute in the URL path.
+
+        Returns:
+            string: The corresponding resource URL.
+        """
+        return self._url_format.format(
+            api=api, project_id=self.project_id, version=self.version, **params)
+
+
+class TenantAwareAuthResourceUrlBuilder(AuthResourceUrlBuilder):
```

Comment:
I didn't see any usages of this class. But seems useful if you ever decide to migrate the multitenancy code as well.

Reply:
Yeah, I ended up reducing all of this cruft into a single function call, I've found it to be much nicer to work with.